common/app/views/fragments/commonCssSetup.scala.html

```diff
@@ -70,21 +50,68 @@
+  - Exclude IE < 10 [(gt IE 9)]
+  - Include IE Mobile [|(IEMobile)]
 *@
-<!--[if (lt IE 9)&(!IEMobile)]>
-    <link rel="stylesheet" type="text/css" href="@Static("stylesheets/old-ie.global.css")" />
-<![endif]-->
+<!--[if (gt IE 9)|(IEMobile)]><!-->
+@if(play.Play.isDev() || !InlineCriticalCss.isSwitchedOn) {
+    <link rel="stylesheet" id="head-css" data-reload="head@projectName.map("." + _).getOrElse(".content")" type="text/css" href="@Static("stylesheets/head" + projectName.map("." + _).getOrElse(".content") + ".css")" />
+} else {
+    <style class="js-loggable">
+        @Html(Static.css.head(projectName))
+    </style>
+}
+    @fragments.stylesheetLink("stylesheets/global.css", "global")
+<!--<![endif]-->
 
-@*
-  - Include IE Mobile [IEMobile]
-  - Note: does not include IE10 on Windows Phone 8 (which does not support conditional comments)
-*@
-<!--[if IEMobile]>
-    <link rel="stylesheet" type="text/css" href="@Static("stylesheets/global.css")" />
-<![endif]-->
+@if(AsyncCss.isSwitchedOn) {
+    <script>
+        // The css is loaded by the `link`s above but with media of "only x".
+        // This means the browser requests it, but doesn't block rendering on it,
+        // since this media is not 'x'. Once it's loaded (appears in
+        // `window.document.styleSheets`),  `media` is set to `all` so the browser
+        // uses it. Very much inspired by https://github.com/filamentgroup/loadCSS.
+        (function (isModernEnough) {
+            if(isModernEnough) {
+                function setMedia (styleSheet, styleSheets) {
+                    styleSheets || (styleSheets = window.document.styleSheets);
+
+                    for (var i = 0, totalSheets = styleSheets.length; i < totalSheets ; i++) {
+                        var sheet = styleSheets[i];
+                        if (sheet.href && sheet.href.indexOf(styleSheet.href) > -1 ) {
+                            styleSheet.media = "screen";
+                            return true;
+                        }
+                    }
+                    setTimeout(function () {
+                        setMedia(styleSheet, styleSheets);
+                    });
+                }
+
+                function useCss () {
+                    var styleSheets = document.querySelectorAll('link[media="only x"]');
+                    for (var i = 0, totalStyleSheets = styleSheets.length; i < totalStyleSheets ; i++) {
+                        setMedia(styleSheets[i]);
+                    }
+                }
+
+                useCss();
+
+                // TEMPORARY: Tidy up localStorage - can remove after a bit
+                try {
+                    Object.keys(localStorage).forEach(function(key) {
+                        if (key.match(/^gu.css./g)) {
+                            localStorage.removeItem(key);
+                        }
+                    });
+                } catch (e) {};
+            }
+        })('querySelectorAll' in document)
```

Comment:
So what do we do for browsers which aren't IE (and get into the conditional statements above) and don't have querySelectorAll support? I.e. do we need and else statement here to just remove/amend the media attribute straight away?